### PR TITLE
fix(agent): sanitize Excel output sheet names

### PR DESCRIPTION
**Pull request by uczltw6**
### Summary

Sanitize `ExcelProcessor` sheet names before writing XLSX output. Multi-sheet agent data can use keys such as `Q1: Sales`; openpyxl rejects `:` and the component currently turns the entire output into an error. This replaces Excel-prohibited characters and titles (including edge apostrophes and the reserved name `History`), preserves the 31-character limit and empty-name fallback, and deduplicates names case-insensitively after normalization.

The regression test writes an actual workbook from invalid, reserved, empty, overlong, and colliding names, reopens it, and verifies the normalized sheets and stored attachment.

Testing:

- `.venv/bin/python -m pytest -q --confcutdir=test/unit_test/agent/component test/unit_test/agent/component/test_excel_processor_output.py` — 1 passed
- `python3 -m ruff check agent/component/excel_processor.py test/unit_test/agent/component/test_excel_processor_output.py` — passed
- `python3 -m ruff format --check agent/component/excel_processor.py test/unit_test/agent/component/test_excel_processor_output.py` — passed
- `.venv/bin/python -m py_compile agent/component/excel_processor.py test/unit_test/agent/component/test_excel_processor_output.py` — passed

OpenAI Codex assisted with identifying the edge case and preparing the focused patch; the behavior and test results were verified locally.

### CODE REVIEW

File: agent/component/excel_processor.py [hunk 1]
```diff
@@ -341,9 +342,17 @@ def _output_excel(self):
...
                         # Sanitize sheet name (max 31 chars, no special chars)
-                        safe_name = sheet_name[:31].replace("/", "_").replace("\\", "_")
+                        base_name = re.sub(r"[\\/*?:\[\]]", "_", str(sheet_name))[:31] or "Sheet"
+                        safe_name = base_name
+                        counter = 2
+                        while safe_name.casefold() in used_sheet_names:
+                            suffix = f"_{counter}"
+                            safe_name = f"{base_name[: 31 - len(suffix)]}{suffix}"
+                            counter += 1
+                        used_sheet_names.add(safe_name.casefold())
```
[the remark this answers is not in the document]
Comment (uczltw6) [reply agent/component/excel_processor.py]: Addressed in 04575e2. The normalization now strips leading/trailing apostrophes, avoids the case-insensitive reserved name `History`, keeps the 31-character and empty-name fallbacks, logs aggregate normalization/deduplication counts, and extends the workbook round-trip test to cover those boundaries.

**Comment by JinHai-CN** [comment]
@Shivamjohri247 Would you please help review this PR?

File: agent/component/excel_processor.py [hunk 2]
```diff
@@ -341,10 +342,32 @@ def _output_excel(self):
...
                 with pd.ExcelWriter(excel_io, engine="openpyxl") as writer:
+                    used_sheet_names = set()
+                    normalized_sheet_count = 0
+                    deduplicated_sheet_count = 0
                     for sheet_name, df in dfs.items():
-                        # Sanitize sheet name (max 31 chars, no special chars)
-                        safe_name = sheet_name[:31].replace("/", "_").replace("\\", "_")
+                        # Sanitize sheet name (max 31 chars, no special or reserved names)
+                        original_name = str(sheet_name)
+                        normalized_name = re.sub(r"[\\/*?:\[\]]", "_", original_name).strip("'")
```
[the remark this answers is not in the document]
Comment (uczltw6) [reply agent/component/excel_processor.py]: Good catch — openpyxl 3.1.5 writes a workbook containing a NUL worksheet title, but the resulting workbook cannot be parsed back. Commit f8847c9 now replaces XML-invalid C0 characters while preserving tab, LF, and CR. The end-to-end regression reopens the generated XLSX and verifies that `Nul\x00Name` becomes `Nul_Name`.

**Comment by uczltw6** [comment]
Addressed the remaining CodeRabbit resource-lifetime note in 82a579e. The regression test now closes the read-only workbook in a `finally` block after inspecting its sheet names. The targeted test, Ruff check/format, py_compile, and diff check pass.

**Review by buua436 — APPROVED** [review]
Reviewed the latest changes. I didn’t find any blocking issues. The worksheet-name sanitization handles invalid/XML characters, empty and overlong names, the reserved History name, and case-insensitive duplicates. The regression test covers these cases and properly closes the read-only workbook. CI is passing.  
Approved.

**Comment by buua436** [comment]
@uczltw6 
 Would you be willing to apply the same fix to the Go implementation as well? Since Go has an equivalent `ExcelProcessor`, it would be great to keep both implementations consistent.
